app/src/services/auth.guard.ts

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { useNavigate } from "react-router-dom";
+
+
+
+export function ProtectedRoute ({ children }: { children: JSX.Element }) {
+  const navigate = useNavigate();
+  if (localStorage.getItem("token")) { 
+    console.log("Token found")
+    return children;
+
+  } else {
+  console.log("No token found")
+
+  React.useEffect(() => {
+    navigate("/")
+  }, []);
+}
+};
+
+export function NonProtectedRoute ({ children }: { children: JSX.Element }) {
+  const navigate = useNavigate();
+  if (!localStorage.getItem("token")) { 
+    return children;
+
+  } else {
+  React.useEffect(() => {
+    navigate("/profile")
+  }, []);
+}
+
+}
+
+
```

Comment:
Functionerne skal have kommentar der følger JSDoc

Reply:
Duly noted: will implement in further sprints.

Reply:
I've made a comment in your backend pull request that explains how the comments should look like. But basically use Doxygen Documentation, if you didn't know😅